lib/middleware/runner.js

```diff
@@ -48,14 +48,20 @@ function createRunnerMiddleware (emitter, fileList, capturedBrowsers, reporter,
         })
       })
 
+      if (config.client.originalArgs) {
+        config.client.args = config.client.originalArgs
+        config.client.originalArgs = null
+      }
       if (_.isEmpty(data.args)) {
         log.debug('Ignoring empty client.args from run command')
       } else if ((_.isArray(data.args) && _.isArray(config.client.args)) ||
         (_.isPlainObject(data.args) && _.isPlainObject(config.client.args))) {
         log.debug('Merging client.args with ', data.args)
+        helper.saveOriginalArgs(config)
```

Comment:
That should never happen because original value of `config.client.args` is restored here https://github.com/karma-runner/karma/pull/3121#discussion-diff-214069975R52

Reply:
Yes, but if we call helper.saveOriginalArgs(config) before we create the handler, we never have to think about the branches here.